plugins/github-pull-requests-board/src/components/EntityTeamPullRequestsContent/EntityTeamPullRequestsContent.tsx

```diff
@@ -36,9 +38,20 @@ const EntityTeamPullRequestsContent = (
 ) => {
   const { pullRequestLimit } = props;
   const [infoCardFormat, setInfoCardFormat] = useState<PRCardFormating[]>([]);
-  const { repositories } = useUserRepositories();
-  const { loading, pullRequests, refreshPullRequests } = usePullRequestsByTeam(
+  const {
+    loading: loadingReposAndTeam,
     repositories,
+    teamMembers,
+    teamMembersOrganization,
+  } = useUserRepositoriesAndTeam();
+  const {
+    loading: loadingPRs,
+    pullRequests,
+    refreshPullRequests,
+  } = usePullRequestsByTeam(
```

Comment:
Just noting that this one will be triggered to load twice - once while waiting for repositories to arrive, and again once they do arrive. Could be that it's easier to make two loader functions `getUserRepositoriesAndTeam` and `getPullRequestsByTeam` and then having a single `useAsync` that calls them one after the other with `await`

Reply:
sounds good, I'll look into it as well

Reply:
@freben For this part, I've drafted something and pushed it to a draft branch: https://github.com/Cylix/backstage/commit/036a2e0a179e8ec71088c1a2c1facf2290ab42f8

Not sure if I'm heading on the right track, but sounds like a loooot of change to extract the `useApi` usage in the hooks that are converted in loaders.

And, this currently break the refresh feature. In the current implementation, it relies on having a `useState` that is updated from a `useCallback`. These have been removed when converting the hooks into loaders, and the only way forward I can see is to re-add a `useState` somewhere, so that sounds like it is somewhat defeating the purpose of what you're suggesting?

Anyway, I'd prefer to double-check with you whether or not I should keep moving forward with that `useAsync` change in the scope of my PR, or just let it go for now 🤔